### PR TITLE
assisted-day2: added periodic and presubmit tests

### PR DESCRIPTION
**Pull request by danielerez**
* Added e2e-metal-assisted-day2-periodic to assisted-test-infra.
* Added e2e-metal-assisted-day2 to assisted-service.

### CODE REVIEW

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml [hunk 1]
```diff
@@ -615,6 +615,16 @@ tests:
         TEST_FUNC=test_olm_operator
       TEST_SUITE: minimal
     workflow: baremetalds-assisted-install
+- as: e2e-metal-assisted-day2
+  optional: true
+  run_if_changed: ^(data/.*)$
```
Comment (eliorerz) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:620]: Why data?

Comment (danielerez) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:620]: looks like most optional tests use it, seems reasonable as the test is too long for running oftenly...

Comment (osherdp) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:620]: Running this job when bumping ocp versions is a good idea
Maybe some other directories, I must say I don't really know which ones are relevant specifically for day2

File: ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml [hunk 2]
```diff
@@ -69,6 +69,76 @@ periodics:
...
+  decorate: true
+  extra_refs:
+  - base_ref: master
+    org: openshift
+    repo: assisted-test-infra
+  labels:
+    ci.openshift.io/generator: prowgen
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-openshift-assisted-test-infra-master-e2e-metal-assisted-day2-periodic
+  spec:
```
Comment (osherdp) [inline ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml:84]:
```suggestion
  reporter_config:
    slack:
      channel: '#assisted-deployment-ci'
      job_states_to_report:
      - failure
      - error
      report_template: '{{ if eq .Status.State "success" }} :green_jenkins_circle:
        {{ else }} :red_jenkins_circle: {{ end }} Job *{{.Spec.Job}}* ended with *{{.Status.State}}*.
        <{{.Status.URL}}|View logs>'
  spec:
```

Comment (osherdp) [reply ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml:84]: this one not being generated by default, so we need to add that to be able to notify failures on slack

**Review by osherdp — CHANGES_REQUESTED** [review]

**Review by osherdp — APPROVED** [review]

**Pull request merged**